lib/Sema/CSSimplify.cpp

```diff
@@ -8097,11 +8137,20 @@ void ConstraintSystem::addExplicitConversionConstraint(
   SmallVector<Constraint *, 3> constraints;
 
   auto locatorPtr = getConstraintLocator(locator);
+  ConstraintLocator *coerceLocator = locatorPtr;
+
+  if (allowFixes && shouldAttemptFixes()) {
+    if (isa<CoerceExpr>(locator.getAnchor()) &&
```

Comment:
Nit: retrieving `anchor` this way from locator builder is has linear complexity, I'd suggest the pointer be stored into a variable once retrieved.